### PR TITLE
`Learning path`: Add learning path explanation for students

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

This PRs introduces students to the concept of learning paths if they did not use it before.

### Description

If a learning path has not been generated yet a new welcome text is displayed. The text shortly introduces the student to the concept of learning paths. Additionally the student generates his/her learning path him/herself by clicking on a new generation button.

### Steps for Testing

Prerequisites:
- 1 student (with no learning path generated yet)
- 1 lecture with learning paths enabled
- at least one Lecture Unit (or Exercise) connected to a competency

1. Log in to Artemis
2. Navigate to the course
3. navigate to learning paths
4. if no learning path has yet been generated a short text should be displayed and a generate button
5. by clicking on the generate button a learning path should be generated and the "normal" learning path UI should be displayed

The following test servers can be used for testing -> Course Johannes Wiest (if you don't want to test locally).
Please mark the user if you tested with it that other testers don't use the same user.
#### [Test Server 1:](https://artemis-test1.artemis.cit.tum.de/)
- [ ] artemis_test_user_2
- [ ] artemis_test_user_4
- [ ] artemis_test_user_5

#### [Test Server 4:](https://artemis-test4.artemis.cit.tum.de/)
- [x] artemis_test_user_2
- [ ] artemis_test_user_3
- [x] artemis_test_user_4
- [x] artemis_test_user_5

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage

image

### CODE REVIEW

**Review by rstief — APPROVED** [review]
tested on ts4 works as expected. Code also lgtm

**Review by edkaya — APPROVED** [review]
Code lgtm

File: src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html [hunk 1]
```diff
@@ -1,12 +1,32 @@
 <div class="col learning-path-student-page">
-    @if (learningPathId()) {
+    @if (isLoading()) {
+        <div class="row justify-content-center align-items-center h-100">
+            <div class="spinner-border text-primary" role="status">
+                <span class="sr-only">Loading...</span>
```
Comment (JohannesStoehr) [inline src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html:5]: Should we add a translation here? We have `Laden...` in other places in Artemis

File: src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+            </div>
+        </div>
+    } @else if (learningPathId()) {
         <jhi-learning-path-student-nav [learningPathId]="learningPathId()!" />
+        <div class="learning-path-student-content">
+            @if (currentLearningObject()?.type === LearningObjectType.LECTURE) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html:11]: Why not use a switch case here? I'm also open to keep it as is

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html (hunk 1 above)
Comment (JohannesWt) [reply src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html:5]: The 'Loading...' is just for accessibility and is never shown. More infos on that here https://getbootstrap.com/docs/5.0/components/spinners/

File: src/test/javascript/spec/component/learning-paths/pages/learning-path-student-page.component.spec.ts [hunk 3]
```diff
@@ -111,4 +112,34 @@ describe('LearningPathStudentPageComponent', () => {
...
+    it('should set isLoading correctly during learning path generation', async () => {
+        jest.spyOn(learningPathApiService, 'getLearningPathId').mockRejectedValue(new EntityNotFoundError());
+        jest.spyOn(learningPathApiService, 'generateLearningPath').mockResolvedValue(learningPathId);
+        const loadingSpy = jest.spyOn(component.isLoading, 'set');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(loadingSpy).toHaveBeenCalledWith(true);
+        expect(loadingSpy).toHaveBeenCalledWith(false);
```
Comment (pzdr7) [inline src/test/javascript/spec/component/learning-paths/pages/learning-path-student-page.component.spec.ts:143]: Just a thought: I wonder if we could be more specific here. Would `toHaveBeenNthCalledWith(1, true)` and `toHaveBeenNthCalledWith(2, false)` work? Same for the test `should set isLoading correctly during learning path load`

This way, we'd be making sure that the order of values is also correct

File: src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html (hunk 1 above)
Comment (pzdr7) [reply src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.html:5]: `sr-only` is for screen readers. A German user would hear an English text here, so I think we should localize it. I found some occurrences that look like this:

```jsx
                <span class="sr-only" jhiTranslate="loading"></span>
```

File: src/main/webapp/i18n/en/learningPath.json [hunk 4]
```diff
@@ -11,6 +11,11 @@
             },
             "competencyGraphModal": {
                 "title": "Learning path competency relations"
+            },
+            "generation": {
+                "title": "Welcome to learning paths!",
+                "description": "Welcome to your learning path! This feature lets you learn at your own pace with a personalized journey tailored to your performance. It adapts dynamically, ensuring you always have the most relevant content and challenges. Start now and experience a customized path to success!",
+                "generateButton": "Start my learning path"
```
Comment (pzdr7) [inline src/main/webapp/i18n/en/learningPath.json:18]: The translations don't seem 100% consistent between German and English here.

For example:

German: Willkommen bei **deinem Lernpfad**
English: Welcome to **learning paths**!

in English, the "Welcome to your learning path" that I would expect in the title (based on the German translation) is in the description instead. 

Is there a reason for this?

**Review by pzdr7 — CHANGES_REQUESTED** [review]

File: src/main/webapp/i18n/en/learningPath.json (hunk 4 above)
Comment (JohannesWt) [reply src/main/webapp/i18n/en/learningPath.json:18]: True I first did the English ones, then thought that the 1to1 translations don't really fit in German. But now I adapted the English ones again. 👍

**Review by pzdr7 — APPROVED** [review]
Code 👍

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by MaximilianAnzinger — APPROVED** [review]
Maintainer approved

**Pull request merged**